temporal-sdk/src/test/java/io/temporal/workflow/DynamicWorkflowTest.java

```diff
@@ -119,4 +126,26 @@ public void testDynamicWorkflowFactory() {
     String result = workflow.getResult(String.class);
     assertEquals("activityType2-activityType1-startArg0-workflowFoo", result);
   }
+
+  @Test
+  public void testDynamicWorkflowFailure() {
```

Comment:
To make it more clean and explicit, it's a good idea to don't catch the exception, add WorkflowFailedException to the test method signature and add it to @Test(expected = WorkflowFailedException.class)